src/_pytest/_code/code.py

```diff
@@ -56,6 +56,15 @@
 
     _TracebackStyle = Literal["long", "short", "line", "no", "native", "value", "auto"]
 
+BaseExceptionGroup: Optional[Type[BaseException]]
```

Comment:
Perhaps we should make `exceptiongroup` a proper dependency in Python<3.11?

This way we can simplify this code quite a bit:

```python
if sys.version_info[:2] < (3, 11):
    from exceptiongroup import BaseExceptionGroup
```

And `BaseExceptionGroup` would always be available.

Reply:
Actually yeah, a hard dependency would also be useful for https://github.com/pytest-dev/pytest/issues/8217

Reply:
Makes sense that it'll be used in more places going forward, fixed :+1:
Also means there's no need for any tests in the vein of `py37-noexceptgroup`